### DISCUSSION TITLE
Length distribution of twin boundaries

### DISCUSSION THREAD

**Discussion by angus0997** [mtex-toolbox/mtex#1606] [Ask Anything] [answered]
Hi all,

I would like to compare the length distribution of the twin boundaries in two samples:

I have EBSD data from two samples, from which I have found the total length of the twin boundaries. These total lengths turn out to be more or less equal to each other.
Now, I would like to look at the length distribution of the twin boundaries and compare them, i.e., does one sample have a higher number of, on average, shorter boundaries, or do the two have a similar distribution of twin boundary lengths. Something like a histogram of Twin boundary length Vs number of boundaries having that length would be useful for me. I hope this makes sense.

Does anyone have suggestions for how to do this?

Thanks,
Angus

**Comment by kilir** [comment, marked as answer]
Hi,
I guess you want something like that?
```
%% prep example data - as in Twinning.m
mtexdata twins

[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',5*degree);
ebsd(grains(grains.grainSize<=2)) = [];

% removing quadruplepoints might be desired
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',5*degree,'removeQuadruplePoints');

grains = grains.smooth(5);

gB_MgMg = grains.boundary('Magnesium','Magnesium')

CS = ebsd('M').CS

twinning = orientation.map(Miller(0,1,-1,-1,CS),Miller(-1,1,0,-1,CS),...
  Miller(1,0,-1,1,CS,'uvw'),Miller(1,0,-1,-1,CS,'uvw'))

isTwinning = angle(gB_MgMg.misorientation,twinning) < 5*degree;
tB = gB_MgMg(isTwinning)

%%

% sum for common componentid
bdprop = accumarray(tB.componentId,tB.segLength);

%repeat for plotting
tBCompLength = bdprop(tB.componentId);
      

plot(grains)
hold on
plot(tB,tBCompLength,'linewidth',2)
hold off
mtexColorbar('title','connected tB length')

%  histogram(bdprop)
```
<img width="557" alt="xmple1" src="https://user-images.githubusercontent.com/11697557/197823128-91f4392a-fa72-4df2-ab3e-ffc2ccde90b3.png">

Cheers,
Rüdiger

**Reply by angus0997** [reply]
Yes - this is exactly what I had in mind, `accumarray(tB.componentId,tB.segLength);` does the job.

Thank you very much :)
